packages/appsync-modelgen-plugin/src/visitors/appsync-visitor.ts

```diff
@@ -221,6 +221,28 @@ type ManyToManyContext = {
   directive: CodeGenDirective;
 };
 
+// Configuration used for processing directives
+export type CodeGenDirectiveProcessConfig = {
+  // This flag is going to be used for using custom primary key feature
+  isCustomPKEnabled: boolean,
+  // This flag is used for native platforms(Android, iOS, Flutter) to use model name field as associatedWith field in implicit bi-dir hasMany
+  shouldUseModelNameFieldInHasManyAndBelongsTo: boolean,
+  // This flag is going to be used to tight-trigger on JS implementations only.
+  shouldImputeKeyForUniDirectionalHasMany: boolean,
+  // This flag is currently used in JS/TS and Model introspection generation only.
+  shouldUseFieldsInAssociatedWithInHasOne: boolean,
+  // This flag is currently used in Swift only to have sort key fields of index included in associatedWith fields
+  shouldRespectSortKeyFieldsOfIndexInAssociatedWithFields:boolean,
+}
+
+export const defaultCodegenDirectiveProcessConfig: CodeGenDirectiveProcessConfig = {
+  isCustomPKEnabled: true,
```

Comment:
The ideal default here would be `this.isCustomPKEnabled()` but since this is defined outside the class, can you specify why we chose `true` as default for this one and not `false` like others below?

Reply:
I just change the default value for CPK to false. This will be override by each visitor's own method of `this.isCustomPKEnabled()`